tests/functional/ui/test_odf_storage_consumption_trend.py

```diff
@@ -0,0 +1,314 @@
+            )
+        elif second_validation:
+            logger.warning(
+                "Manually calculated and UI values are matched with upto 10% difference for 'Estimated days until full'"
+            )
+        assert (
+            first_validation or second_validation
+        ), "'Estimated days until full' is wrongly displayed in UI"
+
+    def test_average_of_storage_consumption_ui(self, setup_ui_class):
+        """
+        Verify the accuracy of ‘Average of storage consumption per day’  in the widget
+            1. Get the value of Average, total storage, age of the cluster
+                and calculate Average of storage consumption.
+            2. Compare the above calculated value with the ‘Average’ in the widget (UI)
+
+        """
+        average = self.get_avg_consumption_from_ui()
+        logger.info(f"From the UI, Average: {average}")
+        days_avg_tpl = self.calculate_est_days_and_average_manually()
+        first_validation = average == days_avg_tpl[1]
+        # rel_tol 0.1 means upto 10% tolerance, which means that difference between  manual and UI est days is up to 10%
+        second_validation = math.isclose(average, days_avg_tpl[1], rel_tol=0.1)
+        if first_validation:
+            logger.info(
+                "Manually calculated and UI values are exactly matched for 'Average'"
+            )
+        elif second_validation:
+            logger.warning(
+                "Manually calculated and UI values are matched with upto 10% difference for 'Average'"
+            )
+        assert (
+            first_validation or second_validation
+        ), "'Average' is wrongly displayed in UI"
+
+    def test_consumption_trend_with_mgr_failover(self, setup_ui_class):
+        """
+        Verify storage consumption trend with Mgr failover
+            1. Failover active mgr pod, the other mgr will become active now.
+            2. Test storage consumption trend from UI is accurate after mgr failover.
+
+        """
+        logger.info("Get mgr pods objs")
+        mgr_objs = get_mgr_pods()
+        toolbox = get_ceph_tools_pod()
+        active_mgr_pod_output = toolbox.exec_cmd_on_pod("ceph mgr stat")
+        active_mgr_pod_suffix = active_mgr_pod_output.get("active_name")
+        logger.info(f"The active MGR pod is {active_mgr_pod_suffix}")
+        active_mgr_deployment_name = "rook-ceph-mgr-" + active_mgr_pod_suffix
+        for obj in mgr_objs:
+            if active_mgr_pod_suffix in obj.name:
+                active_mgr_pod = obj.name
+
+        logger.info(f"Scale down {active_mgr_deployment_name} to 0")
+        helpers.modify_deployment_replica_count(
+            deployment_name=active_mgr_deployment_name, replica_count=0
+        )
+        POD_OBJ.wait_for_delete(resource_name=active_mgr_pod)
+        # Below sleep is mandatory for mgr failover, if not the same pod will become active again.
+        time.sleep(60)
```

Comment:
pls fetch data with TimeoutSampler, this will make test more rigid, fast for execution and notify test engineer in the logs that wait happens

Reply:
+1

Reply:
done